File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/scytale/ScytaleSuite.kt
package cryptobench.ciphers.scytale

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class ScytaleSuite : CipherSuite {
    override fun name(): String = "scytale"

    override fun run(): SuiteResult {
        val cipher: Cipher = ScytaleCipher(ScytaleKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in ScytaleVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/scytale/ScytaleSuite.scala
package cryptobench.ciphers.scytale

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class ScytaleSuite extends CipherSuite {
  override def name: String = "scytale"

  override def run(): SuiteResult = {
    val cipher: Cipher = new ScytaleCipher(ScytaleKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- ScytaleVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
